### PR TITLE
Fix focus issue in VideoOSD (flickering when using mouse)

### PR DESCRIPTION
**Pull request by peak3d**
## Description
When moving mouse while VideoOSD is open, the default focus item (play / pause button) flickers between focused / unfocused.

Reason is that onMouseMove triggers an ACTION_TRIGGER_OSD which opens the OSD and selects the default item.

This PR stops further processing in OSD dialog if it is already open.

## Motivation and Context
Issue was mentioned here and there.

## How Has This Been Tested?
Win10 / play Video / move mouse cursor in bottom third part of the screen (over OSD Dialog area)

## Types of change
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Cosmetic change (non-breaking change that doesn't touch code)
- [ ] None of the above (please explain below)

### CODE REVIEW

File: xbmc/video/windows/GUIWindowFullScreen.cpp [hunk 1]
```diff
@@ -440,6 +440,8 @@ void CGUIWindowFullScreen::TriggerOSD()
   {
     if (!g_application.GetAppPlayer().IsPlayingGame())
       pOSD->SetAutoClose(3000);
+    else if (pOSD->IsActive())
```
[the remark this answers is not in the document]
Comment (fritsch) [reply xbmc/video/windows/GUIWindowFullScreen.cpp]: Another issue I might see. Before the timeout was reset everytime a button was pressed. Now it's not resetted anymore. (Not sure if one should care ... though)

[the remark this answers is not in the document]
Comment (FernetMenta) [reply xbmc/video/windows/GUIWindowFullScreen.cpp]: > Not sure if one should care

of course we should. you don't want the osd being closed while navigating it.

now that the osd dialog eats the message, it has to call SetAutoClose

**Comment by peak3d** [comment]
The early exit is already there:
https://github.com/xbmc/xbmc/blob/master/xbmc/guilib/GUIDialog.cpp#L187-L189

So just FYI one don't have to care about the autoclose at this place.

Nevertheless, this change in GUIFullScreenWindow is not necessary, misinterpreted an action while debugging -> its removed now.

**Comment by FernetMenta** [comment]
The code you linked does not reset the timer on button presses.

**Comment by FernetMenta** [comment]
I think the root case is that the dialog gets opened while still active. We should fix this instead of fighting symptoms.

**Comment by FernetMenta** [comment]
disregard the timer. 
but we still need to fix the root cause. the dialog must not get into this close/reopen state when you move the mouse

**Comment by peak3d** [comment]
It is also triggered by VP in some circumstances.

On some place you'll have to check if the trigger must be send or not, I'll leave it to ppl to figure out better place, I don't use mouse, so I don't care about this PR at all.

T.b.h. That the check itself is done on the place where the dialog itself will be opened again, is fine for me. Why ask for this dialog in lower chain for this? The dialog should open on every mouse move, this is a feature.

Curious what other place will be found.

**Comment by FernetMenta** [comment]
There is a method that resets the timer when mouse is active:
https://github.com/xbmc/xbmc/blob/master/xbmc/video/dialogs/GUIDialogVideoOSD.cpp#L46

if the dialog closes, there is an issue which this PR does hide

**Comment by peak3d** [comment]
@FernetMenta not sure if you have seen that I updated the PR, the changes regarding open() are reverted and are in original state. Only remaining thing is that I don't route TRIGGER_OSD to children of the OSD dialog itself

**Comment by FernetMenta** [comment]
> Only remaining thing is that I don't route TRIGGER_OSD to children of the OSD dialog itself

what difference does this make? children should listen to this action.

**Comment by FernetMenta** [comment]
I looked into it. By eating the action you prevent CGUIWindow::OnAction from getting called.
CGUIWindow::OnAction checks for current focused control and if there is none, it sets default.

This function is obviously wrong and needs fixing. This PT only hides the root cause.

**Pull request closed without merging**